viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -86,19 +100,28 @@ export class FilterTree extends React.Component<FilterTreeProps, FilterTreeState
         let checkedIds: number[] = [];
         const checkedNode = this.getNode(this.props.nodes, id);
         if (checkedNode) {
-            if (checkedNode.children.length) {
-                const childrenIds = this.getAllChildrenIds(checkedNode, []);
-                const visibleChildrenIds = childrenIds.filter((childId: number) =>this.getNode(this.state.filteredNodes, childId) !== undefined);
-                if (this.isNodeChecked(id)) {
-                    checkedIds = checkedIds.concat(visibleChildrenIds);
-                } else {
+            const childrenIds = this.getAllChildrenIds(checkedNode, []);
+            const visibleChildrenIds = childrenIds.filter((childId: number) =>this.getNode(this.state.filteredNodes, childId) !== undefined);
+            if (!this.isNodeChecked(id)) {
```

Comment:
I observe the correct behavior in the application, but reading the code, I fail to understand! In both the if and the else here, there's an if (checkedNode.children.lengt) and a else that does checkedIds = checkedIds.concat(id); Why does it work??

Reply:
behaviour is different if you want to check or uncheck a node so I verify that first. Next is if the node has children or not. I thought it was easier to define it this way

Reply:
I understand, but in both cases, if the node has no children, it gets pushed in the checkedIds? That's the part I don't understand...

Reply:
we need to check or uncheck the leaf, then verify the related nodes status